### PR TITLE
Support multifunction MultiCodeDecoder for Qwen3-TTS (stepped + fused)

### PR DESCRIPTION
**Pull request by EduardoPach**
## Summary

This lands the fused MultiCodeDecoder from @mjfrey's #506. The approach
— collapsing the frame into one call and doing the sampling in-graph with Gumbel-max — is his,
and it's what made this worth shipping. Thanks for the contribution. On our side we've packaged
both graphs into a single multifunction asset, converted the CustomVoice checkpoints, and made
the mode selectable from the SDK.

A talker frame expands into 15 residual codes. Stepped does that in 16 CoreML calls with host
sampling and an embedding lookup between each one; fused does the whole frame in a single call
with the sampling and lookups inside the graph. `.stepped` stays the default, so nothing changes
unless you opt in.

## What's new

- `Qwen3MultiCodeDecoderMode` (`.stepped` / `.fused`) via `TTSKitConfig.multiCodeDecoderMode`,
  mirroring `Qwen3SpeechDecoderMode`.
- CLI: `argmax-cli tts --multi-code-decoder-mode stepped|fused`.
- Example app: a Stepped / Fused picker in the sidebar (flipping it reloads the model).
- Fused Gumbel noise comes from the task's `GreedyTokenSampler` RNG (new
  `TokenSampling.gumbelNoise(count:)`), so `GenerationOptions.seed` reproduces fused audio the
  same way it does on the stepped path.
- `temperature <= 0` uses zero noise at temperature 1, which reduces `argmax(logits/T + G)` to
  `argmax(logits)` — exact greedy.
- Assets converted with a `top_k` input get `GenerationOptions.topK` at runtime instead of
  having k baked in at conversion time.
- Default variant moves from `W8A16` to `W8A16-multifunction`.

## Usage

```swift
// Default: stepped, same output as before
let tts = try await TTSKit()

// Fused
let config = TTSKitConfig(multiCodeDecoderMode: .fused)
let fusedTTS = try await TTSKit(config)
```

```bash
swift run -c release argmax-cli tts \
  --text "The old café on the corner still smelled like rain and roasted coffee." \
  --multi-code-decoder-mode fused --play
```

## Asset download

The default variant changes to `W8A16-multifunction`, so this triggers a new model download.
It's already on [`argmaxinc/ttskit-coreml`](https://huggingface.co/argmaxinc/ttskit-coreml) and
comes down automatically on your next `setupModels()` / first launch — no token, repo override,
or code change needed.

It's a new download rather than a replacement, so the old `W8A16` folder stays on disk:

| Model | old `W8A16` | new `W8A16-multifunction` |
|---|---|---|
| `12hz-0.6b-customvoice` | 110 MB | 144 MB |
| `12hz-1.7b-customvoice` | 113 MB | 178 MB |

One bundle carries both graphs, hence the larger size (trunk weights are shared across the two
functions, so it's well under two separate assets).

Once you're on the new default, the old `W8A16` folder is no longer used and can be deleted to
reclaim the space.

## Performance

Apple M4, 0.6B W8A16, `latencyOptimized` SpeechDecoder, single worker, same text and seed.
Median of 7 runs of ~220 frames each:

| | `.stepped` | `.fused` | |
|---|---|---|---|
| MultiCodeDecoder | 51.5 ms/frame | 35.7 ms/frame | ~1.4× |
| End-to-end | 15.3 steps/s | 19.8 steps/s | +29% |
| Time to first buffer | 100 ms | 80 ms | -20% |

<details>
<summary>Reproduce</summary>

```bash
swift build -c release

T="The old café on the corner still smelled like rain and roasted coffee. Every morning the same man sat by the window with a newspaper he never seemed to finish. The owner knew his order by heart and poured it before he asked. Outside, the tram rattled past on its way to the harbour."

for M in stepped fused; do
  echo "===== $M ====="
  ./.build/release/argmax-cli tts --text "$T" \
    --seed 42 \
    --speech-decoder-mode latencyOptimized \
    --multi-code-decoder-mode $M \
    --concurrent-worker-count 1 \
    --output-path /tmp/mcd_$M --output-format wav --verbose \
    | grep -E "^MultiCodeDecoder:|Steps per Second|Time to first buffer|Total Steps"
done
```

Read `MultiCodeDecoder` (ms/run), `Steps per Second`, and `Time to first buffer` from the timing
block. The table above is the median of 7 runs per mode, alternating the order of the two modes
between rounds; single runs vary by a few ms/frame, and fused is the more variable of the two.

`--concurrent-worker-count 1` forces sequential decoding so per-frame timings are clean, and
`--play` is omitted because playback contention adds noise. The long text is only there to get
~220 frames per run, which averages out per-frame jitter.

Note the two modes don't produce identical audio — fused sampling draws a different (equally
valid) sequence, so the same text comes out as 226 frames stepped vs 219 fused. The figures above
are per-frame and per-step rates, so that difference doesn't distort them.

</details>

## Breaking changes

- If you explicitly pin `multiCodeDecoderVariant: "W8A16"`, drop the override. The loader now
  always selects a function by name, so single-function assets fail to load. Older SDK releases
  pinned to `W8A16` are unaffected — that asset stays published.
- `TokenSampling` gains `gumbelNoise(count:)`, but with a default implementation in a protocol
  extension, so existing conformers still compile. Override it only if you want fused sampling
  to use your own RNG.

---

Co-authored-by: @mjfrey

### CODE REVIEW

**Comment by EduardoPach** [comment]
Screenshot for easy reference of what changed on the UI. Added a mode picker below `SpeechDecoder`'s mode picker

image

File: Sources/TTSKit/Qwen3TTS/Qwen3MultiCodeDecoder.swift [hunk 1]
```diff
@@ -43,17 +43,44 @@ public class Qwen3MultiCodeDecoder: MultiCodeDecoding, @unchecked Sendable {
...
+        do {
+            loaded = try await MLModel.load(contentsOf: url, configuration: modelConfig)
+        } catch {
+            throw TTSError.modelLoadingFailed(
+                "MultiCodeDecoder: failed to load function '\(mode.functionName)' from " +
+                "\(url.lastPathComponent). This must be a multifunction CoreML asset " +
+                "with 'stepped' and 'fused' functions, running on macOS 15 / iOS 18 " +
+                "or newer. (\(error.localizedDescription))"
+            )
+        }
```
[the remark this answers is not in the document]
Comment (EduardoPach) [reply Sources/TTSKit/Qwen3TTS/Qwen3MultiCodeDecoder.swift:75]: Both functions require iOS 18, dismissing.

File: Sources/TTSKit/Qwen3TTS/Qwen3MultiCodeDecoder.swift [hunk 2]
```diff
@@ -75,6 +102,57 @@ public class Qwen3MultiCodeDecoder: MultiCodeDecoding, @unchecked Sendable {
...
+        if options.temperature <= 0 {
+            // Exact greedy: argmax(logits / 1 + 0) == argmax(logits)
+            temperature = 1
+            noise = [FloatType](repeating: 0, count: noiseCount)
+        } else {
+            // The graph divides logits by temperature; keep it strictly positive.
+            temperature = max(options.temperature, 0.05)
+            // Bare `FloatType.init` is ambiguous on x86_64 where `FloatType == Float`.
+            noise = sampler.gumbelNoise(count: noiseCount).map { FloatType($0) }
+        }
```
[the remark this answers is not in the document]
Comment (EduardoPach) [reply Sources/TTSKit/Qwen3TTS/Qwen3MultiCodeDecoder.swift:134]: https://github.com/argmaxinc/argmax-oss-swift/pull/513/commits/3bac16289e095fecc35788b239206ca8a063593d clarified in a comment

File: Sources/TTSKit/Utilities/Sampling.swift [hunk 3]
```diff
@@ -29,6 +29,25 @@ public protocol TokenSampling {
...
+        var rng: any RandomNumberGenerator = SystemRandomNumberGenerator()
+        return Self.gumbelNoise(count: count, using: &rng)
+    }
+
+    static func gumbelNoise(count: Int, using rng: inout any RandomNumberGenerator) -> [Float] {
+        (0..<count).map { _ in
+            // The uniform draw excludes 0 and 1 so both logs stay finite.
+            -log(-log(Float.random(in: Float.leastNormalMagnitude..<1, using: &rng)))
+        }
+    }
```
[the remark this answers is not in the document]
Comment (EduardoPach) [reply Sources/TTSKit/Utilities/Sampling.swift:51]: Added guard, but no precondition https://github.com/argmaxinc/argmax-oss-swift/pull/513/commits/3241565995b82271c99546bdb203c31105fd557d

**Review by a2they — APPROVED** [review]
